### PR TITLE
STYLE: remove unnecessary itkImageFileWriter.h header

### PR DESCRIPTION
**Pull request by issakomi**
Unused and unnecessary _itkImageFileWriter.h_ in _Filtering_ and _Segmentation_ modules.

### CODE REVIEW

**Review by dzenanz — APPROVED** [review]

File: Modules/Filtering/Denoising/include/itkPatchBasedDenoisingImageFilter.hxx [hunk 1]
```diff
@@ -23,7 +23,6 @@
 #include "itkResampleImageFilter.h"
 #include "itkIdentityTransform.h"
 #include "itkLinearInterpolateImageFunction.h"
-#include "itkImageFileWriter.h"
```
Comment (N-Dekker) [inline Modules/Filtering/Denoising/include/itkPatchBasedDenoisingImageFilter.hxx:26]: This `#include "itkImageFileWriter.h"` appears to be introduced with commit 65e90f7017e4c215e9210761c20ea1a21baa03da "ENH: A2D2 Denoising", @krismz, May 30, 2012. But even then, itkPatchBasedDenoisingImageFilter.hxx didn't actually use the header 🤷 Looking at https://github.com/InsightSoftwareConsortium/ITK/blob/65e90f7017e4c215e9210761c20ea1a21baa03da/Modules/Filtering/Denoising/include/itkPatchBasedDenoisingImageFilter.hxx

File: Modules/Segmentation/LevelSets/include/itkVectorThresholdSegmentationLevelSetFunction.hxx [hunk 2]
```diff
@@ -21,7 +21,6 @@
 #include "itkImageRegionIterator.h"
 #include "itkGradientAnisotropicDiffusionImageFilter.h"
 #include "itkLaplacianImageFilter.h"
-#include "itkImageFileWriter.h"
```
Comment (N-Dekker) [inline Modules/Segmentation/LevelSets/include/itkVectorThresholdSegmentationLevelSetFunction.hxx:24]: This `#include "itkImageFileWriter.h"` appears to be introduced with commit e795b8be95b295f4ce38b9cea445b3edadb47db6, "Vector version of the ThresholdSegmentationLevelSet method", Luis Ibanez, Dec 18, 2003. But without actually being used! Looking at
https://github.com/InsightSoftwareConsortium/ITK/blob/e795b8be95b295f4ce38b9cea445b3edadb47db6/Code/BasicFilters/itkVectorThresholdSegmentationLevelSetFunction.txx

File: Modules/Segmentation/LevelSets/include/itkThresholdSegmentationLevelSetFunction.hxx [hunk 3]
```diff
@@ -21,7 +21,6 @@
 #include "itkImageRegionIterator.h"
 #include "itkGradientAnisotropicDiffusionImageFilter.h"
 #include "itkLaplacianImageFilter.h"
-#include "itkImageFileWriter.h"
```
Comment (N-Dekker) [inline Modules/Segmentation/LevelSets/include/itkThresholdSegmentationLevelSetFunction.hxx:24]: This `#include "itkImageFileWriter.h"` appears to be introduced with commit 04f04d55d39a1e6950d7f0e12e2b689ad4cbbaed "ENH: Filter now incorporates an optional Laplacian term...", Josh Cates, July 4, 2003: https://github.com/InsightSoftwareConsortium/ITK/blob/04f04d55d39a1e6950d7f0e12e2b689ad4cbbaed/Code/Algorithms/itkThresholdSegmentationLevelSetFunction.txx

**Review by N-Dekker — APPROVED** [review]
Cool, how did you find these unnecessary `#include "itkImageFileWriter.h"`s? Anyway, approved 👍 of course

**Comment by issakomi** [comment]
>  how did you find these unnecessary `#include "itkImageFileWriter.h"`s?

Thank you. By chance, when I was watching itk::Math PR, then I looked for more.

**Pull request merged**